rewrite-java/src/main/java/org/openrewrite/java/JavaSourceVisitor.java

```diff
@@ -45,32 +46,64 @@
         return getCursor().firstEnclosing(J.ClassDecl.class);
     }
 
-    default boolean isInSameNameScope(Cursor higher, Cursor lower) {
-        AtomicBoolean takeWhile = new AtomicBoolean(true);
-        return higher.getPathAsStream()
+    /**
+     * This method determines if the element referenced by the child cursor has visibility to variables and declarations
+     * that are visible to the element referenced by the base cursor. (Is the child within the lexical scope of the base?)
+     *
+     * The base lexical scope is first established by walking up the path of the base cursor to find it's first enclosing
+     * element. The child path is traversed by walking up the child path elements until either the base scope has
```

Comment:
The term "enclosing element" is used here but not defined. An experienced java programmer who thinks about it will _probably_ infer your intended meaning correctly, but best to be explicit.

Reply:
ok, round three 8)